core-tests/jvm/src/test/scala/zio/RTSSpec.scala

```diff
@@ -81,21 +81,25 @@ object RTSSpec
           assertM(io, isTrue)
         } @@ jvm(nonFlaky(100)),
         testM("deadlock regression 1") {
-          import java.util.concurrent.Executors
+          if (TestVersion.isDotty) {
```

Comment:
We just merged a PR to make tests more lazy so I think this should work with the test aspect now. Can you try after merging upstream changes?